packages/contracts-bedrock/src/L2/L2ToL2CrossDomainMessenger.sol

```diff
@@ -137,6 +165,53 @@ contract L2ToL2CrossDomainMessenger is IL2ToL2CrossDomainMessenger, ISemver {
         msgNonce++;
     }
 
+    /// @notice Sends an unsucessful message hash back to the source chain after a given amount of time has passed.
+    /// @param _source      Chain ID of the original message's source chain.
+    /// @param _nonce       Nonce of the original message.
+    /// @param _sender      Sender of the original message.
+    /// @param _target      Target of the original message.
+    /// @param _message     Message payload of the original message.
+    function sendHashToRollbackInbox(uint256 _source, uint256 _nonce, address _sender, address _target, bytes calldata _message) external {
+        if (_source == block.chainid) revert MessageSourceSameChain();
```

Comment:
is source a required input? could we just have `block.chainid`?

Reply:
This is `_source` as in the chain id of where the message came from (I will change the name to messageSource or something), I use `block.chainid` in the hashing for the messageDestination (the current chain that's sending back the message)

Reply:
oh lol, my bad, you are correct